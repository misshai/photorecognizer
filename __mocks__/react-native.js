import mockCamera from './Camera';

const rn = require('react-native');

jest.mock('Linking', () => {
    return {
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        openURL: jest.fn(),
        canOpenURL: jest.fn(),
        getInitialURL: jest.fn(),
    }
});


jest.mock('ImageStore', () => {
    const ImageStore = jest.fn();
    ImageStore.getBase64ForTag = jest.fn();
    ImageStore.mockSuccess = jest.fn((result)=> {
        ImageStore.getBase64ForTag = jest.fn().mockImplementationOnce((path, success, error)=>success(result));
    });
    ImageStore.mockFailure = jest.fn((errorMessage)=> {
        ImageStore.getBase64ForTag = jest.fn().mockImplementationOnce((path, success, error)=>error(errorMessage));
    });


    return ImageStore;
});


jest.mock('react-native-camera', () => mockCamera);
jest.unmock('redux-mock-store');
jest.unmock('redux-thunk');
module.exports = rn;