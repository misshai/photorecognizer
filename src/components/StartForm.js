import React  from 'react';
import { Card, CardSection, Button } from './common';
import { Actions } from 'react-native-router-flux';

const StartForm = ()=> {
    return (<Card>
        <CardSection>
            <Button onPress={()=>Actions.camera()}>START CAMERA</Button>
        </CardSection>
    </Card>);
}

export default StartForm;