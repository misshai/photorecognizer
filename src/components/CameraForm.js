import React, { Component } from 'react';
import { Text  } from 'react-native';
import { connect } from 'react-redux';
import Camera from 'react-native-camera';
import { Actions } from 'react-native-router-flux';
import { Card, CardSection, Button } from './common';
import { getPictureTitle } from '../actions';

export class CameraForm extends Component {
    styles = {
        container: {
            flex: 1,
            flexDirection: 'row'
        },
        preview: {
            flex: 1,
            justifyContent: 'flex-end',
            alignItems: 'center',
            height: "100%"
        },
        capture: {
            flex: 0,
            backgroundColor: '#fff',
            borderRadius: 5,
            color: '#000',
            padding: 10,
            margin: 40
        }
    };

    render() {
        return (<Card>
            <CardSection >
                <Camera
                    ref={(cam) => {
            this.camera = cam;
          }}
                    style={this.styles.preview}
                    aspect={Camera.constants.Aspect.fill}>
                    <CardSection >
                        <Button onPress={this.takePicture.bind(this)}>CAPTURE</Button>
                    </CardSection>
                </Camera>
            </CardSection>
        </Card>);
    }

    takePicture() {
        const options = {};
        this.camera.capture({metadata: options})
            .then((data) => {
                const {path} = data;
                Actions.result();
                this.props.getPictureTitle({path});

            })
            .catch(err => console.error(err));

    }
}
export default connect(null, {getPictureTitle})(CameraForm);