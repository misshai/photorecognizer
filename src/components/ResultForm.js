import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Text, Image, ScrollView, ListView,View } from 'react-native';
import { Card, CardSection, Button, Spinner } from './common';

export class ResultForm extends Component {
    componentWillMount() {
        this.createDataSource(this.props);
    }

    componentWillReceiveProps(nextProps) {
        this.createDataSource(nextProps);
    }

    createDataSource({ titles }) {
        const ds = new ListView.DataSource({
            rowHasChanged: (r1, r2) => r1 !== r2
        });

        this.dataSource = ds.cloneWithRows(titles);
    }

    render() {
        const {
            containerResultStyle
            } = styles;

        return (<ScrollView>
            <Card>
                <CardSection>
                    {this.displayImage()}
                </CardSection>
                <CardSection style={containerResultStyle}>
                    {this.displayResult()}
                </CardSection>
            </Card>
        </ScrollView>);
    }

    displayResult() {
        if (this.props.error || this.props.titles.length > 0) {
            return (
                <View>
                    <Text style={styles.textErrorStyle}>{this.props.error}</Text>
                    <ListView
                        enableEmptySections
                        dataSource={this.dataSource}
                        renderRow={this.displayLabel}
                    />
                </View>
            );
        } else {
            return <Spinner size='large'/>
        }
    }

    displayImage() {
        if (this.props.imgUrl)
            return (<Image style={styles.imageStyle} source={{ uri: this.props.imgUrl }}/>);
    }

    displayLabel(item) {
        return <Text style={styles.textResultStyle}>{item}</Text>;
    }
}
const styles = {

    containerResultStyle: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
    },
    textErrorStyle: {
        fontSize: 18,
        color: 'red'
    },
    textResultStyle: {
        fontSize: 20,
        color: 'black'
    },

    imageStyle: {
        flex: 1,
        width: null,
        height: 200,
        resizeMode: 'contain'
    }
}

const mapStateToProps = ({ result}) => {
    return {titles: result.titles, imgUrl: result.imgUrl, error: result.error}
}

export default connect(mapStateToProps)(ResultForm);