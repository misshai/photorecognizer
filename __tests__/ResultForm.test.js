import 'react-native';
import React from 'react';
import {ResultForm} from '../src/components/ResultForm';
import renderer from 'react-test-renderer';

test('Result Form', () => {
    const component = renderer.create(
        <ResultForm titles={[]}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});