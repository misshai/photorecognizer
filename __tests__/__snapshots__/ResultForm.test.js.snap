// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Form 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#ddd",
          "borderRadius": 2,
          "borderWidth": 1,
          "elevation": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 10,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 2,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 5,
              "position": "relative",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 5,
              "position": "relative",
            },
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="large"
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
